week1/tictactoe/borboss366/Main.hs

```diff
@@ -0,0 +1,354 @@
+module Main where
+
+import Data.List (groupBy, maximumBy)
+import Data.Function(on)
+import Data.Matrix (matrix, Matrix, ncols, nrows, getElem, setElem)
+import Data.Char(toLower, isAsciiLower, ord, isDigit)
+import System.Console.ANSI (setSGR, ConsoleLayer(..), SGR(..), ColorIntensity(..), Color(..))
+import Text.Read(readMaybe)
+
+newtype TTTPoint = TTTPoint {
+  getPoint :: (Int, Int)
+} deriving (Eq, Show)
+
+newtype TTTDimension = TTTDimension {
+  getDimension :: (Int, Int)
+}  deriving (Eq, Show)
+
+newtype TTTLine = TTTLine {
+  getRow :: [TTTPoint]
+} deriving (Eq, Show)
+
+newtype TTTCellLine = TTTCellLine {
+  getCellRow :: [(TTTCellType, TTTPoint)]
+} deriving (Eq, Show)
+
+data TTTPlayer = Krestig | Nolek deriving (Show, Eq)
+data TTTCellType = Occupied TTTPlayer | EmptyCell deriving (Show, Eq)
+data TTTMoveState = Move TTTPlayer | Win TTTPlayer | TTTie deriving (Eq, Show)
+
+data TTTPosition = TTTPosition {
+  getField :: Matrix TTTCellType,
+  getMoves :: [TTTPoint],
+  get2Win :: Int
```

Comment:
It'd be nice to have comments next to such fields :( Currently it's hard to understand what's going on.

Syntax for Haddock comments:

``` haskell
data TTTPosition = TTTPosition {
  getField :: Matrix TTTCellType,
  getMoves :: [TTTPoint],
  -- | Blah blah this is field for blah blah.
  get2Win :: Int
}
```